gems/pending/Scvmm/miq_hyperv_disk.rb

```diff
@@ -150,7 +150,7 @@ def bread(start_sector, number_sectors)
       encoded_data = @parser.output_to_attribute(run_correct_powershell(read_script))
       t2           = Time.now.getlocal
       buffer       = ""
-      Base64.decode64(encoded_data).split(' ').each { |c| buffer += c.to_i.chr }
+      Base64.decode64(encoded_data).split(' ').each { |c| buffer << c.to_i.chr }
```

Comment:
@roliveri first remember this is not new code.  I believe the reason it is so complicated is that you're using PowerShell to generate the encoded data and Ruby to decode it.  If we were using PowerShell for both ends
it would be one the above command to generate and another to decode.

With regard to splitting on the space:  The output of Base64.decode64(encoded_data) is a character string consisting of string representations of numbers with spaces between each number.  I'm splitting on the space, taking each string number, and using to_i to convert the string to an actual number,  then using .chr to convert it back to the binary encoding.

If there were a "space" in the data - there would be a numeric representation of that space character, not the actual space.

With regard to your second question, yes there is an ASCII.GetString, however, it will not work.  If I replace the GetBytes call with GetString I get the error: "Cannot convert value to System.Byte.  Input string was not in a correct format".